ucdlib-directory/src/public/src/elements/ucdlib-directory-research-highlights.js

```diff
@@ -0,0 +1,231 @@
+import { LitElement } from 'lit';
+import { render } from './ucdlib-directory-research-highlights.tpl.js';
+import { Mixin, LitCorkUtils } from '@ucd-lib/cork-app-utils';
+import { MainDomElement } from "@ucd-lib/theme-elements/utils/mixins/main-dom-element.js";
+
+/**  
+  @classdesc Element to display research highlights for a 
+  directory profile 
+*/
+export default class UcdlibDirectoryResearchHighlights extends Mixin(LitElement) 
+  .with(LitCorkUtils, MainDomElement) {
+  static get properties() {
+    return {
+      expertId: { type: String, attribute: 'expert-id' },
+      res: { type: Array },
+      dataLazy: { type: Boolean, attribute: 'data-lazy', reflect: true },
+      loading: { type: Boolean, state: true },
+      error: { type: String, state: true }
+    };
+  }
+   
+  constructor() {
+    super();
+    this.render = render.bind(this);
+    this.res = [];
+    this.loading = false;
+    this.error = '';
+    this._loadedOnce = false;
+  }
+
+  createRenderRoot() {
+    return this;
+  }
+
+
+  /**  
+    * @description LitElement lifecycle called when element is added to DOM
+    * @returns {void}
+  */
+  connectedCallback() {
+    super.connectedCallback();
+    if (this.hasAttribute('data-lazy')) {
+      this.observeVisibility();
+    } 
+  }
+
+  /**  
+    * @description Fetch research highlights data from API
+    * @returns {void}
+  */
+  async fetchHighlights() {
+    if (this._loadedOnce) return;  
+    if (!this.expertId) return;
+
+    this._loadedOnce = true;
+    this.loading = true;
+    this.error = '';
+    this.res = [];
+
+    try {
+      let url = 'https://experts.ucdavis.edu/api/expert/' + this.expertId;
+      const response = await fetch(url, { headers: { Accept: 'application/json' }});
+
+      if (response.ok) {
+        const raw = await response.text();
+        const data = JSON.parse(raw); 
+        this.res = this.formatResults(data);
+      } else {
+        this.error = `Research highlight ${this.expertId} requested does not exist`;
```

Comment:
`ok` means the status is in the range 200-299: https://developer.mozilla.org/en-US/docs/Web/API/Response/ok

so your error text is not going to always be accurate: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status